chess/src/turn.rs

```diff
@@ -7,21 +7,16 @@ pub struct Turn {
     pub(super) board_state: BoardState,
     pub(super) mv: Move,
     pub(super) piece_captured: bool,
-    pub(super) king_is_checked: bool,
+    pub(crate) status: GameStatus,
 }
 
 impl Turn {
-    pub fn new(
-        board_state: BoardState,
-        mv: Move,
-        piece_captured: bool,
-        king_is_checked: bool,
-    ) -> Self {
+    pub fn new(board_state: BoardState, mv: Move, piece_captured: bool) -> Self {
         Self {
             board_state,
             mv,
             piece_captured,
-            king_is_checked,
```

Comment:
Is there any downside to removing this?

Reply:
It's been replaced with the Game status, so it's more general now